src/test/java/csci432/SigmaDeltaFilterTest.java

```diff
@@ -21,33 +21,94 @@ public void setup() {
     }
 
     @Test
-    public void testRefreshBackground()  {
-        f.refreshBackground(new BufferedImage(300, 300, BufferedImage.TYPE_BYTE_BINARY));
-    }
-
-    @Test
-    public void testSetBackground() {
+    public void testFilter() {
         BufferedImage a = new BufferedImage(300, 300, BufferedImage.TYPE_3BYTE_BGR);
-        f.setBackground(a);
-        f.filter(a);
-        f.setBackground(a);
+        BufferedImage b = a;    //will be calculated background
+        BufferedImage c = a;    //save result here
+        BufferedImage s = a;    //will be calculated result
+
+        for(int i = 0; i < a.getHeight(); i++) {
+            for(int j = 0; j < a.getWidth(); j++) {
+                a.setRGB(j, i, 10);
+                b.setRGB(j, i, 10);
+            }
+        }
+        c = f.filter(a);
+        assertThat(f.background, equalTo(b));
+        assertThat(c,is(equalTo(a)));
+
+        for(int i = 1; i < f.initBackground ; i++){
+            int aPixel = a.getRGB(0,0) + i + 1;
+            int bPixel = b.getRGB(0, 0) + i;
+            for(int y = 0; y < a.getHeight(); y++) {
+                for(int x = 0; x < a.getWidth(); x++) {
+                    a.setRGB(x, y, aPixel);
+                    b.setRGB(x, y, bPixel);
+                    s.setRGB(x, y, aPixel - bPixel);
+                }
+            }
+            c = f.filter(a);
+            assertThat(b, equalTo(f.background));
+            assertThat(c, equalTo(a));
+        }
+
+        int i = f.initBackground;
+        int aPixel = a.getRGB(0,0) + i + 1;
+        int bPixel = b.getRGB(0, 0) + i;
+        for(int y = 0; y < a.getHeight(); y++) {
+            for(int x = 0; x < a.getWidth(); x++) {
+                a.setRGB(x, y, aPixel);
+                b.setRGB(x, y, bPixel);
+            }
+        }
+        c = f.filter(a);
+        assertThat(b, equalTo(f.background));
+        assertThat(c,is(equalTo(s)));
     }
 
     @Test
-    public void testFilter() {
-        BufferedImage a = new BufferedImage(300, 300, BufferedImage.TYPE_3BYTE_BGR);
-        BufferedImage b = f.filter(a);
-        assertThat(b,is(equalTo(a)));
-
-        for(int i = 0; i < 22; i++){
-            b = f.filter(a);
+    public void testRefreshBackground()  {
+        BufferedImage a = new BufferedImage(300, 300, BufferedImage.TYPE_BYTE_GRAY);
+        BufferedImage b = a;
+        BufferedImage c = a;
+        for (int i = 0; i < a.getHeight(); i++){
+            for (int j = 0; j < a.getWidth(); j++){
+                a.setRGB(j, i, 4);
```

Comment:
I don't think this value can possibly be right... I think it is looking for something like 255255255 for white